File: primes.c
#include <stdio.h>
#include <stdlib.h>

int isPrime(int n, int primes[]){
    int p;
    for (int i=0; ; i++){
        p = primes[i];
        if (n % p == 0){
            return 0;
        }
        if (p * p > n) {
            return 1;
        }
    }
}

int * primes(int num){
    int *primes = malloc(sizeof(int) * num);
    primes[0] = 2; primes[1] = 3;
    int found = 2;
    int i = 2;
    int next;
    while (found < num) {
        next = 6 * (i>>1) + (i & 1) * 2 - 1;
        if (isPrime(next, primes)){
            primes[found] = next;
            found++;
        }
        i++;
    }
    return primes;
}

long primesum(int num){
    int *list_of_primes;
    list_of_primes = primes(num);
    int i;
    long sum = 0;
    for (i=0; i<num; i++){
        sum+=list_of_primes[i];
    }
    return sum;
}

int main(int argc, char *argv[]){
    int num;
    num = atoi(argv[1]) * 1000;
    if (num == 0){ num = 100000;}
    long x = primesum(num);
    printf("%ld\n", x);
}

File: primes.java
class Primes {
    private static boolean isPrime(int n, int[] primes) {
        int i = 0;
        while (true) {
            int p = primes[i];
            if ((n%p) == 0) {
                return false;
            }
            if (p*p > n) {
                return true;
            }
            i++;
        }
    }
    private static int[] primes(int n) {
        int[] result = new int[n];
        result[0] = 2; result[1] = 3;
        int found = 2;
        int i = 2;
        int next;
        while (found < n) {
            next = 6 * (i/2) + 2 * (i%2) - 1;
            if (Primes.isPrime(next, result)) {
                result[found] = next;
                found++;
            }
            i++;
        }
        return result;
    }
    private static long primesum(int n) {
        long sum = 0;
        int[] primelist = Primes.primes(n);
        for (int i=0; i<n; i++){
            sum += primelist[i];
        }
        return sum;
    }
    public static void main(String[] args) {
        int n = Integer.parseInt(args[0]) * 1000;
        System.out.println(primesum(n));
    }
}
